File: go2/cmd/esbuild.go
package main

import (
	"fmt"
	"strconv"
	"strings"

	"github.com/evanw/esbuild/pkg/api"
	"github.com/zaydek/retro/color"
)

func FormatTerminalString(msgs []api.Message) string {
	msg := msgs[0] // TODO
	// cwd, _ := os.Getwd()

	return fmt.Sprintf("%s:%d:%d: %s", msg.Location.File, msg.Location.Line, msg.Location.Column, msg.Text) + `

` + color.Boldf("%-*d | %s", len(strconv.Itoa(msg.Location.Line)), msg.Location.Line+0, msg.Location.LineText) + `
` + fmt.Sprintf("%-*d | %s^", len(strconv.Itoa(msg.Location.Line)), msg.Location.Line+1, strings.Repeat(" ", msg.Location.Column)) + `
` + fmt.Sprintf("%-*d | %s%s", len(strconv.Itoa(msg.Location.Line)), msg.Location.Line+2, strings.Repeat(" ", msg.Location.Column), color.BoldRed(msg.Text)) + `
`
}

File: packages/retro/utils/esbuild.ts
import * as esbuild from "esbuild"
import * as term from "../../lib/term"

export function format_esbuild(msg: esbuild.Message, color: (...args: unknown[]) => void): string {
	const meta = msg.location!

	const namespace = `${meta.file}:${meta.line}:${meta.column}`
	const error = `esbuild: ${msg.text}`

	let code = ""
	code += `${meta.lineText.slice(0, meta.column)}`
	code += `${color(meta.lineText.slice(meta.column, meta.column + meta.length))}`
	code += `${meta.lineText.slice(meta.column + meta.length)}`

	return `${namespace}: ${error}

	${meta.line} ${term.dim("|")} ${code}
	${" ".repeat(String(meta.line).length)} \x20 ${" ".repeat(meta.column)}${color("~".repeat(meta.length))}`
}
